test/val/val_extensions_test.cpp

```diff
@@ -118,4 +118,70 @@ TEST_F(ValidateExtensionCapabilities, DeclCapabilityFailure) {
   EXPECT_THAT(getDiagnosticString(), HasSubstr("SPV_KHR_device_group"));
 }
 
+
+using ValidateAMDShaderBallotCapabilities = spvtest::ValidateBase<string>;
+
+TEST_P(ValidateAMDShaderBallotCapabilities, ExpectSuccess) {
+  const string preamble = R"(
+  OpCapability Shader
+  OpCapability Linkage
+  OpExtension "SPV_AMD_shader_ballot"
+  OpMemoryModel Logical GLSL450
+  %float = OpTypeFloat 32
+  %uint = OpTypeInt 32 0
+  %int = OpTypeInt 32 1
+  %scope = OpConstant %uint 3
+  %uint_const = OpConstant %uint 42
+  %int_const = OpConstant %uint 45
+  %float_const = OpConstant %float 3.5
+
+  %void = OpTypeVoid
+  %fn_ty = OpTypeFunction %void
+  %fn = OpFunction %void None %fn_ty
+  %entry = OpLabel
+  )";
+
+  const string assembly = preamble + GetParam() + "\nOpReturn OpFunctionEnd";
+
+  CompileSuccessfully(assembly.c_str());
+  ASSERT_EQ(SPV_SUCCESS, ValidateInstructions()) << getDiagnosticString();
+}
+
+INSTANTIATE_TEST_CASE_P(ExpectSuccess, ValidateAMDShaderBallotCapabilities,
+                        Values(
+                            // clang-format off
```

Comment:
The indentation is strange. Consider putting this before the entire statement.

Reply:
Fixed as a side effect of adding validation failure cases.